ansible/playbooks/tasks/aws-cluster-bootstrap.yaml

```diff
@@ -0,0 +1,404 @@
+  ansible.builtin.command:
+    cmd: crm configure show rsc-options
+  register: crm_rsc_options_show
+  changed_when: false
+  failed_when: false
+
+- name: Get op_defaults cluster configuration
+  ansible.builtin.command:
+    cmd: crm configure show op_defaults
+  register: crm_op_options_show
+  changed_when: false
+  failed_when: false
+
+- name: Set cluster facts
+  ansible.builtin.set_fact:
+    sbd_stonith: "{{ (crm_conf_show.stdout | regex_search('(rsc_iscsi_sbd)', '\\1'))[0] | default('false') }}"
+    stonith_enabled: "{{ (crm_conf_show.stdout | regex_search('stonith-enabled=([a-z]*)', '\\1'))[0] | default('false') }}"
+    stonith_action: "{{ (crm_conf_show.stdout | regex_search('stonith-action=([a-z]*)', '\\1'))[0] | default('false') }}"
+    stonith_timeout: "{{ (crm_conf_show.stdout | regex_search('stonith-timeout=([a-z0-9]*)', '\\1'))[0] | default('false') }}"
+    rsc_ip: "{{ crm_conf_show.stdout | regex_search('(rsc_ip_)') }}"
+    rsc_healthcheck_primary: "{{ crm_conf_show.stdout | regex_search('(rsc_healthcheck_primary)') }}"
+    grp_ip_hc: "{{ crm_conf_show.stdout | regex_search('(grp_ip_hc)') }}"
+    resource_stickiness: "{{ (crm_rsc_options_show.stdout | regex_search('resource-stickiness=([0-9]*)', '\\1'))[0] | default('false') }}"
+    migration_threshold: "{{ (crm_rsc_options_show.stdout | regex_search('migration-threshold=([0-9]*)', '\\1'))[0] | default('false') }}"
+    op_default_timeout: "{{ (crm_op_options_show.stdout | regex_search('timeout=([0-9]*)', '\\1'))[0] | default('false') }}"
+
+- name: Stonith command [AWS]
+  ansible.builtin.set_fact:
+    aws_stonith_cmd: |
+      crm configure primitive rsc_aws_stonith stonith:external/ec2 \
+      op start interval=0 timeout=180 \
+      op stop interval=0 timeout=180 \
+      op monitor interval=120 timeout=60 \
+      meta target-role=Started \
+      params tag={{ aws_stonith_tag}} pcmk_delay_max=15
+
+- name: Enable SBD [sbd]
+  ansible.builtin.command:
+    cmd: crm configure primitive rsc_iscsi_sbd stonith:external/sbd
+  when:
+    - sbd_stonith | string | lower  == 'false'
+    - use_sbd
+    - is_primary
+
+- name: Configure AWS EC2 STONITH for sle 12sp5
+  ansible.builtin.command: "{{ aws_stonith_cmd }}"
+  when:
+    - is_primary
+    - not (use_sbd | bool)
+    - ansible_distribution_major_version == '12'
+  register: stonith_config_result
+  failed_when: >
+    stonith_config_result.stderr_lines | select("match", "ERROR") | reject("match", "ERROR: warning") | list | length > 0
+
+- name: Configure AWS EC2 STONITH for sle 15
+  ansible.builtin.command: "{{ aws_stonith_cmd }}"
+  when:
+    - is_primary
+    - not (use_sbd | bool)
+    - ansible_distribution_major_version != '12'
+  register: stonith_config_result
+  failed_when: "'ERROR' in stonith_config_result.stderr"
+
+# This is a workaround for https://bugzilla.suse.com/show_bug.cgi?id=1231153
+# If it is fixed, this task will no longer be needed
+- name: Install python-httplib2 via zypper
```

Comment:
This task too, it is valid for both GCP and AWS. It has been done as part of https://github.com/SUSE/qe-sap-deployment/pull/279

I propose to move this one (both in this file and GCP) just after task `Ensure cluster dependencies are installed`

Reply:
Sure. I also checked pint (latest release is `sles-12-sp5-sap-v20241028-x86-64`)and the bug (latest update is `[OBSbugzilla Bot](mailto:bwiedemann+obsbugzillabot@suse.com) 2024-11-15 19:50:04 UTC`) so it seems we still need `Install python-httplib2` ATM.